Fix: description of config options

sfc-gh-dmatthews opened the pull request:
`client.showErrorDetails` does not hide the traceback when set to false. This is [expected behavior](https://github.com/streamlit/streamlit/issues/4519#issuecomment-1232782294) currently.

`client.caching` only applies to deprecated `st.cache`. It should be clear that it does not apply to `st.cache_data` and `st.cache_resource`.

`client.toolbarMode` is also affected by Streamlit Community Cloud when set to "auto."

## Describe your changes
Updated description string for configuration option.

## Context
Per @jrieke for `client.showErrorDetails`
Slack: [for client.caching](https://snowflake.slack.com/archives/C039V2A2USJ/p1688132966000539?thread_ts=1688064601.745639&cid=C039V2A2USJ)

**Contribution License Agreement**
By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

sfc-gh-dmatthews committed:
Updated description of client.showErrorDetails config option

client.showErrorDetails does not hide the traceback when set to false. It's description was updated to correctly reflect the current behavior.

snehankekre commented:
@sfc-gh-dmatthews I believe this behavior is a [bug](https://github.com/streamlit/streamlit/pull/5945), not a feature. The tracebacks are supposed to be hidden on the browser when `client.showErrorDetails` is set to False. Probably a good idea to close this PR and submit a [bug report](https://github.com/streamlit/streamlit/issues/new?assignees=&labels=type%3Abug%2Cstatus%3Aneeds-triage&projects=&template=bug_report.yml) instead.

sfc-gh-dmatthews commented:
I asked @jrieke about this, but I could have misunderstood. He said he was going to take on a project next quarter to change it. I had understood from #4519 that it was intentional.

snehankekre commented:
Thanks for the context!

sfc-gh-dmatthews committed:
Updated description of client.caching config option

Added explicit note that the client.caching config option does not apply to the new cache functions (st.cache_data and st.cache_resource).

snehankekre commented on lib/streamlit/config.py:
```diff
@@ -465,9 +467,10 @@ def _logger_message_format() -> str:
+        If set to False, exceptions will display in the browser with the
+        exception type and traceback, along with a generic error message.
+        Deprecation warnings and full exception messages will be printed to the
+        console only.""",
```
Let's reorder these sentences:

```suggestion
        If set to False, deprecation warnings and full exception messages 
        will be printed to the console only. Exceptions will still display in the 
        browser with the exception type and traceback, along with a generic 
        error message.""",
```

snehankekre committed:
Update lib/streamlit/config.py

jrieke commented:
Oh actually @sfc-gh-dmatthews, I just realized we should change the wording slightly here to point out that we'll change this behavior in the future. Maybe something like:

    If set to False, deprecation warnings and full exception messages 
    will be printed to the console only. Exceptions will still display in the 
    browser with a generic error message. For now, the exception type and traceback
    are shown as well but they will be removed in the future. 

And we should also show that in the docs. Just to make sure that people don't rely on this behavior right now and then they feel that we "break" them once we change it.

sfc-gh-dmatthews commented:
Ok, I'll add that it will be changed in the future. (I was just about to push another edit.) For the docs, I just have all the definitions copied from core, so I will definitely update it in docs if I change the definition here.

sfc-gh-dmatthews committed:
Update config.py

Clarified that `showErrorDetails` will be updated in the future and that `client.toolbarMode= "auto"` also depends on Community Cloud.

sfc-gh-dmatthews committed:
Remove whitespace

Removed trailing whitespace

sfc-gh-dmatthews committed:
Removed whitespace

Removed trailing whitespace

lukasmasuch reviewed (approved):
LGTM 👍
